torch/distributed/fsdp/_limiter_utils.py

```diff
@@ -0,0 +1,33 @@
+import collections
+from typing import Deque, Optional
+
+import torch
+
+
+class _FreeEventQueue:
+    """
+    This tracks all pending frees corresponding to inflight all-gathers. The
+    queueing pattern is iterative enqueues with a single dequeue per iteration
```

Comment:
what does `iteration` here refer to?  An FSDP unit is an "iteration"?

Reply:
Yes, one `FlatParameter` is the unit of iteration here.